pkg/ccip/codec/msghasher_test.go

```diff
@@ -0,0 +1,154 @@
+package codec
+
+import (
+	"context"
+	"math/big"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	chainsel "github.com/smartcontractkit/chain-selectors"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+
+	mocks "github.com/smartcontractkit/chainlink-ton/mocks/ccipocr3"
+)
+
+// Extract a single message from the executecodec_test.go helper
+func randomTONMessage(t *testing.T, sourceChainSelector uint64) ccipocr3.Message {
+	report := randomTONExecuteReport(t, sourceChainSelector)
+	return report.ChainReports[0].Messages[0]
+}
+
+func TestMessageHasherV1_TON(t *testing.T) {
+	ctx := context.Background()
+	mockExtraDataCodec := new(mocks.SourceChainExtraDataCodec)
+	edc := ccipocr3.ExtraDataCodec(map[string]ccipocr3.SourceChainExtraDataCodec{
+		chainsel.FamilyEVM: mockExtraDataCodec,
+	})
+
+	mockExtraDataCodec.On("DecodeDestExecDataToMap", mock.Anything).Return(map[string]any{
+		"destgasamount": uint32(1000),
+	}, nil)
+	mockExtraDataCodec.On("DecodeExtraArgsToMap", mock.Anything).Return(map[string]any{
+		"gasLimit": big.NewInt(1000),
+	}, nil)
+
+	lg := logger.Test(t)
+	hasher := NewMessageHasherV1(lg, edc)
+
+	t.Run("successful hash generation", func(t *testing.T) {
+		msg := randomTONMessage(t, 5009297550715157269)
+		hash, err := hasher.Hash(ctx, msg)
+		require.NoError(t, err)
+		assert.NotEqual(t, [32]byte{}, hash)
```

Comment:
To test for hashing correctness you should generate test data via the typescript tests and confirm the produced message_id hash matches

Reply:
Just added it, however it failed and I noticed the Any2TVM message type in the [contract](https://github.com/smartcontractkit/chainlink-ton/blob/c8fd71a67ab7f24c1a2194d65c5fcd3fda1f2953/contracts/contracts/ccip/types.tolk#L45) doesn't have gasLimit but the codec binding [does](https://github.com/smartcontractkit/chainlink-ton/blob/c8fd71a67ab7f24c1a2194d65c5fcd3fda1f2953/pkg/ccip/bindings/ocr/executereport.go#L28)